### PR TITLE
initial stub on lassie integration

### PR DESCRIPTION
**Pull request by willscott**
* it looked like the range check was being overly aggressive. okay to scope it down like this?
* thoughts on the proposed header check for when to respond from lassie vs gateway?

### CODE REVIEW

**Comment by willscott** [comment]
> I feel like this is breaking the HTTP spec a bit. 

that was making it harder for some mis-coded browser client to accidentally send that header value. that said, happy to do it as a `x-...` header or similar. what would you prefer for name / value/

**Comment by guanzo** [comment]
> that was making it harder for some mis-coded browser client to accidentally send that header value. 

ya good point. it also makes it harder for well-coded browser clients to intentionally send the header value.  🤔 

ok spit balling here but how about `x-rhea` or `x-fetcher` for the name, and `lassie` as the value. Then if we want to A/B test new lassie versions or something different altogether, we can add more values.

**Comment by willscott** [comment]
`x-fetcher: bifrost-gateway/1.0.0`?

**Comment by guanzo** [comment]
sgtm

**Comment by guanzo** [comment]
Shall we communicate whenever the `bifrost-gateway` version is bumped? Or just do a prefix match.

**Review by guanzo — APPROVED** [review]

File: container/shim/src/index.js [hunk 1]
```diff
@@ -118,6 +118,11 @@ const handleCID = asyncHandler(async (req, res) => {
 
   debug(`Cache miss for ${req.path}`);
 
+  if ((req.headers["x-fetcher"] || "").indexOf("bifrost-gateway") > -1) {
```
Comment (juliangruber) [inline container/shim/src/index.js:121]:
```diff
-   if ((req.headers["x-fetcher"] || "").indexOf("bifrost-gateway") > -1) {
+   if ((req.headers["x-fetcher"] || "").includes("bifrost-gateway")) {
```

Bystander comment: This should be easier to read

Comment (willscott) [reply container/shim/src/index.js:121]: 😅  my JS is a couple years out of date

Comment (guanzo) [reply container/shim/src/index.js:121]:
```js
if (req.headers["x-fetcher"]?.includes("bifrost-gateway"))`
```

prolly the most l33t hacker version

Comment (juliangruber) [reply container/shim/src/index.js:121]: Nice!

**Pull request merged**